ui/src/alerting/actions/notifications/endpoints.ts

```diff
@@ -101,10 +104,47 @@ export const createEndpoint = (endpoint: NotificationEndpoint) => async (
     type: 'SET_ENDPOINT',
     endpoint: resp.data,
   })
+
+  dispatch(notify(copy.createEndpointSuccess()))
   dispatch(checkEndpointsLimits())
 }
 
-export const updateEndpoint = (endpoint: NotificationEndpoint) => async (
+export const testExistingEndpoint = async (endpoint: NotificationEndpoint) => {
+  const labels = endpoint.labels || []
+
+  const data = {
+    ...endpoint,
+    labels: labels.map(l => l.id),
+  } as PostNotificationEndpoint
+
+  const resp = await api.putNotificationEndpointsTest({
+    endpointID: endpoint.id,
+    data,
+  })
+
+  if (resp.status !== 204) {
+    throw new Error('Test unsuccessful. Check endpoint parameters')
+  }
+}
+
+export const testEndpoint = (endpoint: NotificationEndpoint) => async () => {
+  const labels = endpoint.labels || []
+
+  const data = {
+    ...endpoint,
+    labels: labels.map(l => l.id),
+  } as PostNotificationEndpoint
+
+  const resp = await api.postNotificationEndpointsTest({data})
+
+  if (resp.status !== 204) {
```

Comment:
It doesn't seem like `testEndpoint` ends being used in some instances, would it be more beneficial to dispatch a notification to the user?